R/rowClust.R

```diff
@@ -0,0 +1,83 @@
+#' Clusters rowData using \code{fastcluster::hclust}
+#'
+#' \code{rowClust} Performs hierarchical clustering on rowData of the 
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}, 
+#' by using the dissimilarity function provided at \code{dissimilarity_FUN}.
+#'
+#' @param x a
+#'   \code{\link[SummarizedExperiment:SummarizedExperiment-class]{SummarizedExperiment}}
+#'   object.
+#'   
+#' @param abund_values a \code{character} vector, name of the assay containing the data.
+#' 
+#' @param name a \code{character} vector, name of the new column at \code{rowData} where to place cluster ids.
+#' 
+#' @param clustering_method a \code{character} vector representing the clustering 
+#' method to be used, same as methods available at \code{\link[hclust]{hclust}}.
+#' 
+#' @param transposed \code{logical} whether the data should be transposed.
```

Comment:
Is this necessary? The assays are _always_ having "taxa x samples" format, if the SE object has been constructed correctly.

Reply:
From the initial commit how I tested the calculation of the dissimilarity matrix, correlation and cosine needed the assay transposed. However, wen using stats::dist() the assay didn't need to be transposed. ([link](https://github.com/microbiome/mia/commit/c4eb8e0e14771ba529281caf7c8b016349581f72)) (line 20-25).
Therefore, I thought that the user, could define how the assay should be handled (e.g. dissimilarity_FUN=function(x) cor(t(x))) , or the function would transpose it? (error will be thrown in a dim mismatch case) 
What do you think is the best?

Reply:
Ah - that's a tricky one. 

I feel that it would be safer if we require that the user-provided function is always assuming the same format (features x samples). Then the user is responsible for respecting these. Having an additional `transpose` argument is probably more confusing, and transpose is easy to add to user-defined function as well, if necessary.